src/synapse/workload.py

```diff
@@ -823,3 +823,41 @@ def get_environment(charm_state: CharmState) -> typing.Dict[str, str]:
             environment[proxy_variable] = str(proxy_value)
             environment[proxy_variable.upper()] = str(proxy_value)
     return environment
+
+
+def generate_nginx_config(container: ops.Container, main_unit_address: str) -> None:
+    """Generate NGINX configuration based on templates.
+
+    1. Copy template files as configuration files to be used.
+    2. Run sed command to replace string main-unit in configuration files.
+    3. Reload NGINX.
+
+    Args:
+        container: Container of the charm.
+        main_unit_address: Main unit address to be used in configuration.
+    """
+    container.exec(
```

Comment:
Out of curiosity, may I ask why .template was used instead of jinja? I'm asking because it might be more Pythonic to replace strings than doing sed operations :D

Reply:
Excellent suggestion! I admit that sed was my first idea so I didn't consider Jinja. I created a story to do that as a refactor: ISD-1745
Thanks!